platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -0,0 +1,820 @@
+                    "Invalid AuthnRequest, defined an AssertionConsumerServiceURL, but contained no identifying signature.");
+        }
+
+        if (authnRequest.getRequestedAuthnContext() != null) {
+            Collection authNContextClasses = CollectionUtils.transformedCollection(
+                    authnRequest.getRequestedAuthnContext().getAuthnContextClassRefs(),
+                    new AuthnContextClassTransformer());
+            if (authnRequest.isPassive() && authnRequest.getRequestedAuthnContext().getComparison()
+                    .equals(AuthnContextComparisonTypeEnumeration.EXACT) && !CollectionUtils
+                    .containsAny(authNContextClasses,
+                            Arrays.asList(SAML2Constants.AUTH_CONTEXT_CLASS_REF_X509,
+                                    SAML2Constants.AUTH_CONTEXT_CLASS_REF_SMARTCARD_PKI,
+                                    SAML2Constants.AUTH_CONTEXT_CLASS_REF_SOFTWARE_PKI,
+                                    SAML2Constants.AUTH_CONTEXT_CLASS_REF_SPKI,
+                                    SAML2Constants.AUTH_CONTEXT_CLASS_REF_TLS_CLIENT))) {
+                throw new IllegalArgumentException(
+                        "Unable to passively log user in when not specifying PKI AuthnContextClassRef");
+            }
+        }
+
+        if (!(authnRequest.getProtocolBinding().equals(HTTP_POST_BINDING) || authnRequest
+                .getProtocolBinding().equals(HTTP_REDIRECT_BINDING))) {
+            throw new UnsupportedOperationException(
+                    "Only HTTP-POST and HTTP-Redirect bindings are supported");
+        }
+    }
+
+    protected void validateRelayState(String relayState) {
+        LOGGER.debug("Validating RelayState");
+        if (relayState == null) {
+            throw new IllegalArgumentException("Missing RelayState on IdP request.");
+        }
+        if (relayState.getBytes().length < 0 || relayState.getBytes().length > 80) {
+            LOGGER.warn("RelayState has invalid size: {}", relayState.getBytes().length);
+        }
+    }
+
+    @GET
+    @Path("/sso")
+    public Response processLogin(@QueryParam(SAML_REQ) String samlRequest,
+            @QueryParam(RELAY_STATE) String relayState, @QueryParam(AUTH_METHOD) String authMethod,
+            @QueryParam(SSOConstants.SIG_ALG) String signatureAlgorithm,
+            @QueryParam(SSOConstants.SIGNATURE) String signature,
+            @Context HttpServletRequest request) {
+        LOGGER.debug("Processing login request: [ authMethod {} ], [ sigAlg {} ], [ relayState {} ]", authMethod, signatureAlgorithm, relayState);
+        try {
+            validateRelayState(relayState);
+            AuthnRequest authnRequest = unwrapAuthnRequest(samlRequest, false);
+            validateAuthnRequest(authnRequest, samlRequest, relayState, signatureAlgorithm,
+                    signature,
+                    (StringUtils.isEmpty(signatureAlgorithm) || StringUtils.isEmpty(signature)));
```

Comment:
This may be correct, but it reads very strangely. This is a `GET` op, so it's definitely not a post. Maybe a rename of the param on the `validateAuthnRequest` method would make it less confusing.